pkg/parser/slurm.go

```diff
@@ -183,3 +198,185 @@ func GpusFlag(val string) (map[string]*resource.Quantity, error) {
 
 	return gpus, nil
 }
+
+func SlurmValidateAndReplaceScript(script string, nTasks int32) (string, error) {
+	if nTasks < 1 {
+		return "", fmt.Errorf("invalid number of tasks: %d", nTasks)
+	}
+
+	var (
+		sb          strings.Builder
+		srunCount   int
+		line        string
+		needNewLine bool
+	)
+
+	lines := strings.Split(script, "\n")
+
+	for i := 0; i < len(lines); i++ {
+		line = ""
+
+		// Convert multiline command to one line
+		for i < len(lines) {
+			if len(line) > 0 {
+				line += " "
+			}
+			line += strings.TrimSpace(lines[i])
+			if !strings.HasSuffix(line, " \\") {
+				break
+			}
+			line = strings.TrimRight(line[:len(line)-2], " ")
+			i++
+		}
+
+		if line == "" {
+			sb.WriteByte('\n')
+			continue
+		}
+
+		if nTasks > 1 && srunCount > 0 {
+			return "", errors.New("multiple parallel tasks with a task count greater than 1 are not supported yet")
+		}
+
+		if index := findCommand(line, srunCommand); index != -1 {
+			srunCount++
+
+			args := strings.Split(line[index:], " ")[1:]
+
+			srunFlagSet := pflag.NewFlagSet(srunCommand, pflag.ContinueOnError)
+			srunFlagSet.ParseErrorsWhitelist.UnknownFlags = true
+
+			err := srunFlagSet.Parse(args)
+			if err != nil {
+				return "", fmt.Errorf("invalid %q command: %w", line, err)
+			}
```

Comment:
I think we may have a potential issue here. If a user decides to use quotes in their arguments like: `srun -flag1 "arguments for flag 1"` and/or `srun -flag1 'arguments for flag 1'`. Maybe we could modify the `splitBySpaceWithIgnoreInQuotes` function to support this case or create something similar?

Reply:
Oh, this is a good case, thanks. Fixed on 20de503.